src/components/MultiHopTrade/components/SharedTradeInput/SharedTradeInput.tsx

```diff
@@ -0,0 +1,122 @@
+
+import { SharedTradeInputBody } from '../SharedTradeInput/SharedTradeInputBody'
+import { SharedTradeInputHeader } from '../SharedTradeInput/SharedTradeInputHeader'
+import { ConfirmSummary } from '../TradeInput/components/ConfirmSummary'
+import { WithLazyMount } from '../TradeInput/components/WithLazyMount'
+import { useSharedHeight } from '../TradeInput/hooks/useSharedHeight'
+
+type SharedTradeInputProps = {
+  activeQuote: TradeQuote | undefined
+  buyAmountAfterFeesCryptoPrecision: string | undefined
+  buyAmountAfterFeesUserCurrency: string | undefined
+  buyAsset: Asset
+  hasUserEnteredAmount: boolean
+  headerRightContent: JSX.Element
+  initialBuyAssetAccountId: AccountId | undefined
+  initialSellAssetAccountId: AccountId | undefined
+  isCompact: boolean | undefined
+  isLoading: boolean
+  manualReceiveAddress: string | undefined
+  sellAsset: Asset
+  sideComponent: React.ComponentType<any>
+  tradeInputRef: React.RefObject<HTMLDivElement>
+  walletReceiveAddress: string | undefined
+  handleSwitchAssets: () => void
+  onChangeTab: (newTab: TradeInputTab) => void
+  onSubmit: (e: FormEvent<unknown>) => void
+  setBuyAsset: (asset: Asset) => void
+  setBuyAssetAccountId: (accountId: string) => void
+  setSellAsset: (asset: Asset) => void
+  setSellAssetAccountId: (accountId: string) => void
+}
+
+export const SharedTradeInput: React.FC<SharedTradeInputProps> = ({
+  activeQuote,
+  buyAmountAfterFeesCryptoPrecision,
+  buyAmountAfterFeesUserCurrency,
+  buyAsset,
+  hasUserEnteredAmount,
+  headerRightContent,
+  initialBuyAssetAccountId,
+  initialSellAssetAccountId,
+  isCompact,
+  isLoading,
+  manualReceiveAddress,
+  sellAsset,
+  sideComponent,
+  tradeInputRef,
+  walletReceiveAddress,
+  handleSwitchAssets,
+  onChangeTab,
+  onSubmit,
+  setBuyAsset,
+  setBuyAssetAccountId,
+  setSellAsset,
+  setSellAssetAccountId,
+}) => {
+  const totalHeight = useSharedHeight(tradeInputRef)
+  const [isSmallerThanXl] = useMediaQuery(`(max-width: ${breakpoints.xl})`, { ssr: false })
+
+  return (
+    <Flex
+      id='test-flex'
+      width='full'
+      justifyContent='center'
+      maxWidth={isCompact || isSmallerThanXl ? '500px' : undefined}
+    >
+      <Center width='inherit'>
+        <Card
+          flex={1}
+          width='full'
+          maxWidth='500px'
+          ref={tradeInputRef}
+          as='form'
+          onSubmit={onSubmit}
+        >
+          <SharedTradeInputHeader
+            initialTab={TradeInputTab.Trade}
+            rightContent={headerRightContent}
+            onChangeTab={onChangeTab}
+          />
+          <SharedTradeInputBody
+            activeQuote={activeQuote}
+            buyAmountAfterFeesCryptoPrecision={buyAmountAfterFeesCryptoPrecision}
+            buyAmountAfterFeesUserCurrency={buyAmountAfterFeesUserCurrency}
+            buyAsset={buyAsset}
+            initialBuyAssetAccountId={initialBuyAssetAccountId}
+            initialSellAssetAccountId={initialSellAssetAccountId}
```

Comment:
suggestion: while these are called *initial* in destructuring of `useAccountId()` return value here:

https://github.com/shapeshift/web/blob/8272547a1fb78a64d436c0489dda52e32defdda3/src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx#L82-L87

There is not much of `initial`:

https://github.com/shapeshift/web/blob/253ed2c6d25c3a3c36b9613d22ba4d9f995d4d1e/src/components/MultiHopTrade/hooks/useAccountIds.tsx#L15-L17

I got quite confused while working in this domain, as this is really the *selected* AccountIds, do we want to use this terminology instead as part of the refactor?